### PR TITLE
OSASINFRA-3465: openstack: Update Gophercloud to v2

### PR DESCRIPTION
**Pull request by pierreprinetti**
This PR includes two commits:
* the first commit replaces `Generate` with `GenerateWithContext` in some assets, so that a context is made available for the platform-specific logic.
* the second commit upgrades the Installer to Gophercloud v2.0.0-rc.3. Gophercloud v2 requires a context to be passed to each function; this PR wires the context of each OpenStack API call to the asset-context. Several other minor changes are needed to adapt to the breaking changes in the library.

### CODE REVIEW

File: pkg/asset/manifests/cloudproviderconfig.go [hunk 1]
```diff
@@ -109,7 +110,7 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 		cm.Data[cloudProviderConfigDataKey] = `[Global]
 `
 	case openstacktypes.Name:
-		cloudProviderConfigData, cloudProviderConfigCABundleData, err := openstackmanifests.GenerateCloudProviderConfig(*installConfig.Config)
+		cloudProviderConfigData, cloudProviderConfigCABundleData, err := openstackmanifests.GenerateCloudProviderConfig(context.TODO(), *installConfig.Config)
```
Comment (gryf) [inline pkg/asset/manifests/cloudproviderconfig.go:113]: shouldn't be just `ctx`?

Comment (pierreprinetti) [reply pkg/asset/manifests/cloudproviderconfig.go]: absolutely! Thanks.

**Comment by pierreprinetti** [comment]
Rebased.

**Comment by pierreprinetti** [comment]
Converting to draft to avoid CI resource consumption until next week

**Comment by patrickdillon** [comment]
/approve

For peace of mind let's try basic tests on all platforms
/test e2e-azure-ovn e2e-gcp-ovn e2e-nutanix-ovn e2e-vsphere-ovn e2e-ibmcloud-ovn e2e-powervs-ovn

**Pull request merged**